internal/pubsub/mempubsub/mem.go

```diff
@@ -183,7 +180,7 @@ func (s *subscription) ReceiveBatch(ctx context.Context) ([]*driver.Message, err
 	// alternative would be complicated by the need to recognize expired messages
 	// promptly.
 	for {
-		if msgs := s.receiveNoWait(time.Now(), maxBatchSize); len(msgs) > 0 {
+		if msgs := s.receiveNoWait(time.Now(), maxMessages); len(msgs) > 0 {
```

Comment:
You need to fix `receiveNoWait` to honor `maxMessages`. Although I'm not sure why any of this is in this PR.

Reply:
You're right, I should move this stuff into a separate PR.

Reply:
Moved it to #775 .